src/luxonis_ml/data/utils/enums.py

```diff
@@ -0,0 +1,59 @@
+from enum import Enum
+
+
+class DataLabelType(Enum):
+    """Supported computer vision label types.
+    Annotation types can be nested
+    (e.g. a BOX has 2 LABELS,
+        a BOX has a POLYLINE instance segmentation,
+        etc.)"""
+
+    CLASSIFICATION = (
+        "classification"  # used for single, multi-class, or multi-label classification
+    )
+    BOX = "box"  # bounding box
+    POLYLINE = "polyline"  # polyline to represent segmentation mask instances
+    KEYPOINTS = "keypoints"  # keypoint skeleton instances
+    LABEL = "label"  # an arbitrary label of string, bool, or number
```

Comment:
Could we add a suppord for 3D array like label type that can be then used for for example heatmap labels and similar 2D based image-like labels? I would have it as 3D so it also supports multiple "masks" (e.g. heatmap for each keypoint class). And I would put as a separate type (opposed to merging it to LABEL) so it's easier to manage for augmentations - these would go through image augmentations (e.g. rotate, crop,..) where as LABEL wouldn't as those are "1D types"

Reply:
This could also apply to segmentation maps as well (as opposed to polyline), right? I like the idea, but it might be interesting to test whether we should store these in parquet files or numpy array (or the like) with just paths to these arrays in annotation `value`.

I will add it as an upcoming feature, thanks!